### PR TITLE
[P3 Bug] Update pokemon-info-container ability highlighting to match tinted ball ability check

### PR DESCRIPTION
**Pull request by Tempo-anon**
## What are the changes the user will see?
Makes the caught pokemon-info-container ability highlighting work off the same logic as the tinted ball ability check

## Why am I making these changes?
This was never merged

## What are the changes from a developer perspective?
Created the `checkIfPlayerHasAbilityOfStarter` helper function that is called in both the tinted ball logic as well as the newly caught pokemon logic

### Screenshots/Videos
n/a

## How to test the changes?
Pull and test

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by flx-sta** [comment]
@Tempo-anon mind adding some video/screenshots?
I don't know what the pokemon-info-container is at the top of my head

**Comment by Tempo-anon** [comment]
IMG_7624
This is the pokemon-info-container that shows up when you catch something. This PR just makes the logic for highlighting the ability as blue or not the same as the one used in the tinted pokeball ability check

File: src/ui/pokemon-info-container.ts [hunk 1]
```diff
@@ -267,18 +267,13 @@ export default class PokemonInfoContainer extends Phaser.GameObjects.Container {
...
-      const opponentPokemonAbilityIndex = (opponentPokemonOneNormalAbility && pokemon.abilityIndex === 1) ? 2 : pokemon.abilityIndex;
-      const opponentPokemonAbilityAttr = 1 << opponentPokemonAbilityIndex;
-
-      const rootFormHasHiddenAbility = starterEntry.abilityAttr & opponentPokemonAbilityAttr;
-
-      if (!rootFormHasHiddenAbility) {
+
+      const ownedAbilityAttrs = pokemon.scene.gameData.starterData[pokemon.species.getRootSpeciesId()].abilityAttr;
+
+      // Check if the playe rowns ability for the root form
```
Comment (MokaStitcher) [inline src/ui/pokemon-info-container.ts:273]:
```suggestion
      // Check if the player owns ability for the root form
```
nit

**Review by MokaStitcher — APPROVED** [review]
lgtm

**Pull request merged**